### PR TITLE
6433 fix infinite spinners

### PR DESCRIPTION
**Pull request by mrsarm**
# Description

- Fix infinity spinner issue described here: medic/cht-core#6433
- Also add IntelliJ-based IDEs settings rule to .gitignore

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by mrsarm** [comment]
Now if you try to access to the "Messages" or "Contacts" tab and there are connection issues with the CHT server, you will see the spinner disappear and replaced by the following message box:

Screenshot_20200707_130935

File: webapp/src/js/controllers/messages.js [hunk 1]
```diff
@@ -84,7 +84,12 @@ angular
           });
         }
       })
-      .catch(err => $log.error('Error fetching contact', err));
+      .catch(function(err) {
+        $log.error('Error fetching contact', err);
+        ctrl.loading = false;
+        ctrl.conversations = [];
+        ctrl.error = 'Error fetching contact';
```
Comment (garethbowen) [inline webapp/src/js/controllers/messages.js:91]: The error is rendered by [this code](https://github.com/medic/cht-core/blob/master/webapp/src/templates/directives/messages_list.html#L20) which just checks if it's truthy or not. I think this should just be...

```suggestion
        ctrl.error = true;
```

File: webapp/src/js/controllers/contacts.js [hunk 2]
```diff
@@ -299,6 +299,12 @@ const PAGE_SIZE = 50;
         };
         setActionBarData();
         return ctrl.search();
+      })
+      .catch(err => {
+        $log.error('Error initializing contacts controller', err);
+        ctrl.error = true;
+        ctrl.loading = false;
+        ctrl.appending = false;
```
Comment (garethbowen) [inline webapp/src/js/controllers/contacts.js:307]: Did you manage to reproduce this error somehow?

Comment (mrsarm) [reply webapp/src/js/controllers/contacts.js:307]: Kind of... previous to this patch the "Contacts" tab was displaying the error message but if you tried to reload the all page you got the infinite spinner along with many parts of the UI not working well:

Screenshot_20200708_123129

Now with the patch at least an error is shown but still you see the lack of components that are not directly related with the patch:

Screenshot_20200708_120950

**Pull request merged**